core/routemgmt/common/apigw-utils.js

```diff
@@ -24,6 +24,177 @@ var _ = require('lodash');
+ * is correct.
+ *
+ * @param swaggerApi   Required. The API swagger object to send to the API gateway
+ * @return A promise with the fully validated swaggerApi, or an error response if rejected.
+ */
+function validateFinalSwagger(swaggerApi) {
+  return new Promise(function(resolve, reject) {
+    // This returns a map of urls to check for path parameters.
+    console.log("validateFinalSwagger: Validating swapper before posting to API GW.")
+    var errorMsg;
+    var paths = swaggerApi['paths'];
+
+    if (swaggerApi.basePath && isParameterizedPath(swaggerApi.basePath)) {
+      errorMsg = "The base path (" + swaggerApi.basePath + ") cannot have parameters. Only the relative path supports path parameters.";
+    }
+    /*
+     * This code will look at each path defined, and look at all the parameters in each path, and validate that each
+     * verb (GET,POST, etc) for each path defines parameter objects for each parameter defined in the path. For each of
+     * these that contain parameters, it will also check that its target-url ends in $(request.path).
+     * #beginPathValidation
+     */
+    var targetUrlMap = generateTargetUrlMap(swaggerApi['x-ibm-configuration']);
+    for (var key in paths) {
+      if (errorMsg) { break; }
+      var idx = 0;
+      if (isParameterizedPath(key)) {
+        if (!isValidRelativePath(key)) {
+          errorMsg = "Relative path '" + key + "' does not include valid path parameters.  "
+          errorMsg += "Each parameter must be enclosed in curly braces{}.";
+          break;
+        }
+        //Path is valid, lets check that we have parameters defined for each path parameter and that the target-url
+        //has $(request.path) at the end.
+        var namedParamsInPath = getPathParameters(key);
+        //Loop over each verb (GET,POST,etc), each should contain path parameters.
+        for (var httpType in paths[key]) {
+          var parameters = paths[key][httpType].parameters;
+          var opId = paths[key][httpType].operationId;
+          var xOpenWhisk = paths[key][httpType]['x-openwhisk']
+          //Look at all the parameters in each verb to ensure they match those defined in the path.
+          if (namedParamsInPath == null || parameters == null || namedParamsInPath.length != parameters.length) {
+            var namedLen = namedParamsInPath != null ? namedParamsInPath.length: null;
+            var paramLen = parameters != null ? parameters.length : null;
+            errorMsg = "The number of parameters defined (" + paramLen + ") under HTTP method '" + httpType;
+            errorMsg += "' does not match the number of parameters (" + namedLen
+            errorMsg += ") defined in the path (" + key + ")";
+            break;
+          }
+          if (xOpenWhisk && xOpenWhisk['url'] && !xOpenWhisk['url'].endsWith('.http')) {
+            errorMsg = "The action used use a response type of '.http' in order to receive the path parameters.";
```

Comment:
the error message should say something like... "The action response type must be .http"